src/app/task_manager.h

```diff
@@ -0,0 +1,229 @@
+// LibreSprite
+// Copyright (C) 2021 LibreSprite contributors
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 2 as
+// published by the Free Software Foundation.
+
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <unordered_set>
```

Comment:
nitpicking: LibreSprite files have external libraries declared after the project headers